### ISSUE TITLE
Prompt Sequence 

### ISSUE THREAD

**Issue by rcraggs** [sjwall/MaterialTapTargetPrompt#115]
For an application I'm writing I've written some code that allows you to do something like this (example written in kotlin but would work in Java):

```kotlin
val sequence = MaterialTapTargetSequence()
sequence.add(MaterialTapTargetPrompt.Builder(this).setTaget(target1)) // create a new prompt
sequence.add(MaterialTapTargetPrompt.Builder(this).setTaget(target2)) // create a new prompt
// ...etc...
sequence.show()
```
This allows you to string prompts together in a sequence, simplifying what was described in #31 

Would you be interested in a PR that added this to the library?

**Comment by sjwall** [comment]
Yes I would, it has one of the things that I have wanted to add. Something like you describe would be good

**Comment by rcraggs** [comment]
Thanks,

I've started on this. My initial implementation worked by adding a `setPromptStateChangeListener` to each of the prompts as they were added to the sequence. This would mean that any existing listener that the user had added to the prompt before adding it to the sequence would be removed.

I'll have a think about the best solution to this. Current ideas are:

1. Allow PromptOptions to have a list of state change listeners. The problem with this is to avoid a breaking change, what would `setPromptStateChangeListener` do? E.g. if a user adds a prompt to a sequence and then changes the listener before running the sequence?

2. Provide a package level _additional_ listener to `PromptOption` which can only be set by other code in the library which is used by the sequence to set up the next call. 

I don't think either of these options is great (although option one, if done right may open up a range of possibilities for multiple listeners). I'll have a ponder...

**Comment by rcraggs** [comment]
Initial version will require the user to tap through the sequence, not support for timers yet. Will need to think about that.

**Comment by rcraggs** [comment]
Added PR - #118 

Needs more tests but I don't know yet how to add them for Roboelectric

**Comment by sjwall** [comment]
Thanks for your work, I am reviewing it atm. The aim behind the `extras` package is to have it contain anything additional beyond getting the tap target working as described in the material design guidelines. This hopefully makes it easier for new users of the library to get setup

**Comment by sjwall** [comment]
This has been released in `v2.9.0`
